File: cpp/src/ev1/ut3_strings/ejercicios/pendientes/E09_InvertirCadena_Pendiente.cpp
/*
OBJETIVO: Lee una cadena e imprime la misma cadena invertida con un bucle.
Este ejercicio ahora se ejecuta desde un menu do-while con opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
#include <string>
using namespace std;

int main() {
    string opcion;

    do {
        cout << "=== E09 Invertir cadena (pendiente) ===" << endl;
        cout << "1. Intentar ejercicio" << endl;
        cout << "2. Ver objetivo" << endl;
        cout << "0. Salir" << endl;
        cout << "Elige una opcion: ";
        getline(cin, opcion);
        cout << endl;

        if (opcion == "1") {
            cout << "Escribe una cadena: ";
            string texto;
            getline(cin, texto);

            cout << "Instrucciones:" << endl;
            cout << "- Recorre la cadena desde el final hasta el principio." << endl;
            cout << "- Ve concatenando cada caracter en otra variable con +=." << endl;
            cout << "- En EV1 evita arrays y punteros." << endl;

            string invertida = "";

            // TODO: construye la cadena invertida con un for descendente.

            cout << "Invertida: " << invertida << endl;
        } else if (opcion == "2") {
            cout << "Objetivo: invertir una cadena usando un bucle simple y sin arrays." << endl;
        } else if (opcion == "0") {
            cout << "Saliendo del ejercicio." << endl;
        } else {
            cout << "Opcion no valida." << endl;
        }

        cout << endl;
    } while (opcion != "0");

    return 0;
}

File: csharp/src/ev1/ut3_strings/ejercicios/pendientes/E09_InvertirCadena_Pendiente.cs
/*
OBJETIVO: Lee una cadena e imprime la misma cadena invertida con un bucle.
Este ejercicio ahora se ejecuta desde un menu do-while con opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

string opcion;

do
{
    Console.WriteLine("=== E09 Invertir cadena (pendiente) ===");
    Console.WriteLine("1. Intentar ejercicio");
    Console.WriteLine("2. Ver objetivo");
    Console.WriteLine("0. Salir");
    Console.Write("Elige una opcion: ");
    opcion = Console.ReadLine() ?? "0";
    Console.WriteLine();

    switch (opcion)
    {
        case "1":
            Console.Write("Escribe una cadena: ");
            string texto = Console.ReadLine() ?? "";

            Console.WriteLine("Instrucciones:");
            Console.WriteLine("- Recorre la cadena desde el final hasta el principio.");
            Console.WriteLine("- Ve concatenando cada caracter en otra variable.");
            Console.WriteLine("- En EV1 evita arrays.");

            string invertida = "";

            // TODO: construye la cadena invertida con un for descendente.

            Console.WriteLine("Invertida: " + invertida);
            break;

        case "2":
            Console.WriteLine("Objetivo: invertir una cadena usando un bucle simple y sin arrays.");
            break;

        case "0":
            Console.WriteLine("Saliendo del ejercicio.");
            break;

        default:
            Console.WriteLine("Opcion no valida.");
            break;
    }

    Console.WriteLine();
} while (opcion != "0");
